cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -134,6 +109,122 @@ public final class ServerAuthenticationController: ServerAuthenticationControlli
         #endif
     }
 
+    private func cliManagedAuthenticationToken(serverURL: URL, forceRefresh: Bool) async throws -> AuthenticationToken? {
+        return try await cachedValueStore
+            .getValue(key: "token_\(serverURL.absoluteString)") { () -> (value: AuthenticationToken, expiresAt: Date?)? in
+                guard let token = try await fetchTokenFromStore(serverURL: serverURL) else { return nil }
+
+                let upToDateToken: AuthenticationToken
+                var expiresAt: Date?
+
+                switch token {
+                case .project:
+                    upToDateToken = token
+                case let .user(
+                    legacyToken: legacyToken, accessToken: accessToken, refreshToken: refreshToken
```

Comment:
I wonder if we should get rid of the `legacyToken` – I doubt there's anybody still using that one and even if they are, we might wanna just force them to reauthenticate.